cycletracker/javascript/app.js

```diff
@@ -0,0 +1,90 @@
+// JS file for cycleTracker app
+
+// -------------
+// Variable declarations
+// -------------
+const newPeriodFormEl = document.getElementsByTagName("form")[0];
+const startDateInputEl = document.getElementById("start-date");
+const endDateInputEl = document.getElementById("end-date");
+const pastPeriodContainer = document.getElementById("past-periods");
+
+// Storage key is an app-wide constant
+const STORAGE_KEY = "period-tracker";
+
+// -------------
+// Event Handlers
+// -------------
+newPeriodFormEl.addEventListener("submit", (event) => {
+  event.preventDefault();
+  const startDate = startDateInputEl.value;
+  const endDate = endDateInputEl.value;
+  if (checkDatesInvalid(startDate, endDate)) {
+    return;
+  }
+  storeNewPeriod(startDate, endDate);
+  renderPastPeriods();
+  newPeriodFormEl.reset();
+});
+
+// -------------
+// Functionality
+// -------------
+
+// 1. Form validation
+function checkDatesInvalid(startDate, endDate) {
+  if (!startDate || !endDate || startDate > endDate) {
+    newPeriodFormEl.reset();
+    return true;
+  }
+}
+
+// 2. Get, add, sort, and store data
+function storeNewPeriod(startDate, endDate) {
+  const periods = getAllStoredPeriods();
+  periods.push({ startDate, endDate });
+  periods.sort((a, b) => {
+    return new Date(b.startDate) - new Date(a.startDate);
+  });
+  window.localStorage.setItem(STORAGE_KEY, JSON.stringify(periods));
+}
+
+// 3. Get and parse data
+function getAllStoredPeriods() {
+  const data = window.localStorage.getItem(STORAGE_KEY);
+  const periods = data ? JSON.parse(data) : [];
+  return periods;
+}
+
+// 4. Display data
+function renderPastPeriods() {
+  const pastPeriodHeader = document.createElement("h2");
+  const pastPeriodList = document.createElement("ul");
+  const periods = getAllStoredPeriods();
+  if (Object.keys(periods).length === 0) {
```

Comment:
Isn't this an array, so can't we just do:

```suggestion
  if (periods.length === 0) {
```

??